pretyping/inductiveops.ml

```diff
@@ -456,16 +456,20 @@ let make_case_invert env sigma (IndType (((ind,u),params),indices)) ~case_releva
   then Constr.CaseInvert {indices=Array.of_list indices}
   else Constr.NoInvert
 
+let error_not_allowed_dependent_analysis env isrec i =
+  let open Pp in
+  str "Dependent " ++ str (if isrec then "induction" else "case analysis") ++
+  strbrk " is not allowed for " ++ Termops.pr_global_env env (IndRef i) ++ str "." ++
+  str "Primitive records must have eta conversion to allow dependent elimination."
```

Comment:
Why not merge this with the algebraic error from Indrec?

Reply:
Because inv.ml uses the error from indrec for flow control and I don't want to mess with it.